zksync_sdk/wallet.py

```diff
@@ -253,17 +303,52 @@ async def build_transfer(self, to: str, amount: Decimal, token: TokenLike,
     async def transfer(self, to: str, amount: Decimal, token: TokenLike,
                        fee: Decimal = None,
                        valid_from=DEFAULT_VALID_FROM, valid_until=DEFAULT_VALID_UNTIL) -> str:
-        transfer, eth_signature = await self.build_transfer(to, amount, token, fee,
-                                                            valid_from, valid_until)
+        nonce = await self.zk_provider.get_account_nonce(self.address())
+        token = await self.resolve_token(token)
+
+        if fee is None:
+            fee = await self.zk_provider.get_transaction_fee(FeeTxType.transfer, to, token.id)
+            fee = fee.total_fee
+        else:
+            fee = token.from_decimal(fee)
 
+        amount = token.from_decimal(amount)
+
+        transfer, eth_signature = await self.build_transfer(to, amount, token, fee, nonce, valid_from, valid_until)
         return await self.send_signed_transaction(transfer, eth_signature)
 
+    async def transfer_nft(self, to: str, nft: NFT, fee_token: TokenLike, 
+                          fee: Decimal = None, 
+                          valid_from=DEFAULT_VALID_FROM, 
+                          valid_until=DEFAULT_VALID_UNTIL
+                          ) -> List[TxEthSignature]:
+        nonce = await self.zk_provider.get_account_nonce(self.address())
+        fee_token = await self.resolve_token(fee_token)
+
+        if fee is None:
+            fee = await self.zk_provider.get_transactions_batch_fee(
+                [FeeTxType.transfer, FeeTxType.transfer], 
+                [to, self.address()], 
+                fee_token.symbol
+            )
+        else:
+            fee = fee_token.from_decimal(fee)
+        
+        nft_tx = await self.build_transfer(to, 1, nft, 0, nonce, valid_from, valid_until)
+        fee_tx = await self.build_transfer(self.address(), 0, fee_token, fee, nonce+1, valid_from, valid_until)
+        batch = [
+            TransactionWithSignature(nft_tx[0], nft_tx[1]),
+            TransactionWithSignature(fee_tx[0], fee_tx[1])
+        ]
+        return await self.send_txs_batch(batch)
+
     async def get_order(self, token_sell: TokenLike, token_buy: TokenLike,
                         ratio: Fraction, ratio_type: RatioType, amount: Decimal,
                         recipient: str = None,
+                        nonce: int = None,
                         valid_from=DEFAULT_VALID_FROM,
                         valid_until=DEFAULT_VALID_UNTIL) -> Order:
-        account_id, nonce = await self.zk_provider.get_account_nonce(self.address())
+        nonce = await self.zk_provider.get_account_nonce(self.address())
```

Comment:
This nonce assignment probably shouldn't be here, since we are checking if nonce is None later